docs_src/microservices/security/edgex-core-ssh-proxy.yml

```diff
@@ -0,0 +1,585 @@
+#     ports:
+#     - "127.0.0.1:49990:49990"
+#     container_name: edgex-device-virtual
+#     hostname: edgex-device-virtual
+#     networks:
+#       edgex-network:
+#         aliases:
+#         - edgex-device-virtual
+#     environment:
+#       <<: *common-variables
+#       Service_Host: edgex-device-virtual
+#     depends_on:
+#       - consul
+# #      - logging  # uncomment if re-enabled remote logging
+#       - data
+#       - metadata
+
+  # device-rest:
+  #   image: edgexfoundry/docker-device-rest-go:1.1.0
+  #   ports:
+  #     - "127.0.0.1:49986:49986"
+  #   container_name: edgex-device-rest
+  #   hostname: edgex-device-rest
+  #   networks:
+  #     edgex-network:
+  #       aliases:
+  #         - edgex-device-rest
+  #   environment:
+  #     <<: *common-variables
+  #     Service_Host: edgex-device-rest
+  #   depends_on:
+  #     - data
+  #     - command
+  #      - logging  # uncomment if re-enabled remote logging
+
+  # device-random:
+  #   image: edgexfoundry/docker-device-random-go:1.2.0
+  #   ports:
+  #     - "127.0.0.1:49988:49988"
+  #   container_name: edgex-device-random
+  #   hostname: edgex-device-random
+  #   networks:
+  #     - edgex-network
+  #      aliases:
+  #      - edgex-device-random
+  #   environment:
+  #     <<: *common-variables
+  #     Service_Host: edgex-device-random
+  #   depends_on:
+  #     - data
+  #     - command
+  #
+  # device-mqtt:
+  #   image: edgexfoundry/docker-device-mqtt-go:1.2.0
+  #   ports:
+  #     - "127.0.0.1:49982:49982"
+  #   container_name: edgex-device-mqtt
+  #   hostname: edgex-device-mqtt
+  #   networks:
+  #     - edgex-network
+  #      aliases:
+  #      - edgex-device-mqtt
+  #   environment:
+  #     <<: *common-variables
+  #     Service_Host: edgex-device-mqtt
+  #   depends_on:
+  #     - data
+  #     - command
+  #
+  # device-modbus:
+  #   image: edgexfoundry/docker-device-modbus-go:1.2.0
+  #   ports:
+  #     - "127.0.0.1:49991:49991"
+  #   container_name: edgex-device-modbus
+  #   hostname: edgex-device-modbus
+  #   networks:
+  #     - edgex-network
+  #      aliases:
+  #      - edgex-device-modbus
+  #   environment:
+  #     <<: *common-variables
+  #     Service_Host: edgex-device-modbus
+  #   depends_on:
+  #     - data
+  #     - command
+  #
+  # device-snmp:
+  #   image: edgexfoundry/docker-device-snmp-go:1.2.0
+  #   ports:
+  #     - "127.0.0.1:49993:49993"
+  #   container_name: edgex-device-snmp
+  #   hostname: edgex-device-snmp
+  #   networks:
+  #     - edgex-network
+  #      aliases:
+  #      - edgex-device-snmp
+  #   environment:
+  #     <<: *common-variables
+  #     Service_Host: edgex-device-snmp
+  #   depends_on:
+  #     - data
+  #     - command
+
+##########################################################
+# ssh tunneling proxy service
+##########################################################
+  device-ssh-proxy:
+    image: device-ssh-proxy:test
+    volumes:
+      - $HOME/.ssh:/root/ssh:ro
```

Comment:
Let's not use the .ssh key of the logged on user...  Have the user create a keypair specifically for device forwarding.